moveit_fake_controller_manager/src/moveit_fake_controllers.cpp

```diff
@@ -0,0 +1,288 @@
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Ioan Sucan, Robert Haschke */
+
+#include "moveit_fake_controllers.h"
+#include <ros/param.h>
+#include <sensor_msgs/JointState.h>
+#include <boost/thread.hpp>
+
+namespace moveit_fake_controller_manager
+{
+
+BaseFakeController::BaseFakeController(const std::string &name, const std::vector<std::string> &joints, const ros::Publisher &pub)
+  : moveit_controller_manager::MoveItControllerHandle(name)
+  , joints_(joints)
+  , pub_(pub)
+{
+  std::stringstream ss;
+  ss << "Fake controller '" << name << "' with joints [ ";
+  std::copy(joints.begin(), joints.end(), std::ostream_iterator<std::string>(ss, " "));
+  ss << "]";
+  ROS_INFO_STREAM(ss.str());
+}
+
+void BaseFakeController::getJoints(std::vector<std::string> &joints) const
+{
+  joints = joints_;
+}
+
+moveit_controller_manager::ExecutionStatus BaseFakeController::getLastExecutionStatus()
+{
+  return moveit_controller_manager::ExecutionStatus::SUCCEEDED;
+}
+
+
+LastPointController::LastPointController(const std::string &name,
+                                         const std::vector<std::string> &joints,
+                                         const ros::Publisher &pub) :
+  BaseFakeController(name, joints, pub)
+{
+}
+
+LastPointController::~LastPointController()
+{
+}
+
+bool LastPointController::sendTrajectory(const moveit_msgs::RobotTrajectory &t)
+{
+  ROS_INFO("Fake execution of trajectory");
+  if (t.joint_trajectory.points.empty())
+    return true;
+
+  sensor_msgs::JointState js;
+  const trajectory_msgs::JointTrajectoryPoint &last = t.joint_trajectory.points.back();
+  js.header = t.joint_trajectory.header;
+  js.header.stamp = ros::Time::now();
+  js.name = t.joint_trajectory.joint_names;
+  js.position = last.positions;
+  js.velocity = last.velocities;
+  js.effort   = last.effort;
+  pub_.publish(js);
+
+  return true;
+}
+
+bool LastPointController::cancelExecution()
+{
+  return true;
+}
+
+bool LastPointController::waitForExecution(const ros::Duration &)
+{
+  ros::Duration(0.5).sleep(); // give some time to receive the published JointState
```

Comment:
Hm. It was there before. I removed it and it failed to update to the new pose.
However, https://github.com/ros-planning/moveit_ros/pull/716 it might be superfluous now.